COMETwebapp/Components/ReferenceData/CategoriesTable.razor.cs

```diff
@@ -120,5 +105,29 @@ protected override Task OnInitializedAsync()
 
             return base.OnInitializedAsync();
         }
+
+        /// <summary>
+        /// Method invoked after each time the component has been rendered. Note that the component does
+        /// not automatically re-render after the completion of any returned <see cref="Task"/>, because
+        /// that would cause an infinite render loop.
+        /// </summary>
+        /// <param name="firstRender">
+        /// Set to <c>true</c> if this is the first time <see cref="OnAfterRender(bool)"/> has been invoked
+        /// on this component instance; otherwise <c>false</c>.
+        /// </param>
+        /// <returns>A <see cref="Task"/> representing any asynchronous operation.</returns>
+        /// <remarks>
+        /// The <see cref="OnAfterRender(bool)"/> and <see cref="OnAfterRenderAsync(bool)"/> lifecycle methods
+        /// are useful for performing interop, or interacting with values received from <c>@ref</c>.
+        /// Use the <paramref name="firstRender"/> parameter to ensure that initialization work is only performed
+        /// once.
+        /// </remarks>
+        protected override async Task OnAfterRenderAsync(bool firstRender)
+        {
+            await base.OnAfterRenderAsync(firstRender);
+
+            this.Disposables.Add(this.WhenAnyValue(x => x.ViewModel.showHideDeprecatedThingsService.ShowDeprecatedThings)
```

Comment:
Put that on the OnInitializedAsync method. Moreover, the OnInitializedAsync does not include any Async context, preferable to overrive the OnInitialized method instead

Reply:
The Grid is no referenced on initialization, i will have a NullReferenceException

Reply:
Surround with an if(firstRender) then. Currently, on each render, you add a subscription